Tests/ArrayDiffingTests.m

```diff
@@ -121,12 +131,20 @@ - (void)testDiffingInsertionsAndDeletions {
   for (NSArray *test in tests) {
     NSIndexSet *insertions, *deletions;
     [test[0] asdk_diffWithArray:test[1] insertions:&insertions deletions:&deletions];
+    NSMutableIndexSet *mutableInsertions = [insertions mutableCopy];
+    NSMutableIndexSet *mutableDeletions = [deletions mutableCopy];
+
     for (NSNumber *index in (NSArray *)test[2]) {
-      XCTAssert([insertions containsIndex:[index integerValue]]);
+      XCTAssert([mutableInsertions containsIndex:[index integerValue]]);
+      [mutableInsertions removeIndex:[index integerValue]];
     }
     for (NSNumber *index in (NSArray *)test[3]) {
-      XCTAssert([deletions containsIndex:[index integerValue]]);
+      XCTAssert([mutableDeletions containsIndex:[index integerValue]]);
+      [mutableDeletions removeIndex:[index integerValue]];
     }
+
+    XCTAssert([mutableInsertions count] == 0, @"Unaccounted insertions: %@", mutableInsertions);
```

Comment:
Without these, the test actually didn't fail when I tried deleting one of the supposed-to-be-present index values from the test set.

So although everything seems to behave correctly, this was important to ensure the validation checks both necessity and sufficiency of the test set.